zebra/redistribute.c

```diff
@@ -478,14 +477,13 @@ void zebra_interface_address_delete_update(struct interface *ifp,
 
 	zebra_vxlan_add_del_gw_macip(ifp, ifc->address, 0);
 
-	router_id_del_address(ifc);
```

Comment:
This ordering change may cause some issues. Did you try to test any race condition scenarios for this ?